packages/components/src/tab-panel/test/index.tsx

```diff
@@ -49,289 +54,404 @@ describe( 'TabPanel', () => {
 		window.HTMLElement.prototype.getClientRects = originalGetClientRects;
 	} );
 
-	it( 'should render a tabpanel, and clicking should change tabs', async () => {
-		const user = setupUser();
-		const panelRenderFunction = jest.fn();
-		const mockOnSelect = jest.fn();
-
-		render(
-			<TabPanel
-				tabs={ TABS }
-				children={ panelRenderFunction }
-				onSelect={ mockOnSelect }
-			/>
-		);
-
-		expect( getSelectedTab() ).toHaveTextContent( 'Alpha' );
-		expect(
-			screen.getByRole( 'tabpanel', { name: 'Alpha' } )
-		).toBeInTheDocument();
-		expect( panelRenderFunction ).toHaveBeenLastCalledWith( TABS[ 0 ] );
-		expect( mockOnSelect ).toHaveBeenLastCalledWith( 'alpha' );
-
-		await user.click( screen.getByRole( 'tab', { name: 'Beta' } ) );
-
-		expect( getSelectedTab() ).toHaveTextContent( 'Beta' );
-		expect(
-			screen.getByRole( 'tabpanel', { name: 'Beta' } )
-		).toBeInTheDocument();
-		expect( panelRenderFunction ).toHaveBeenLastCalledWith( TABS[ 1 ] );
-		expect( mockOnSelect ).toHaveBeenLastCalledWith( 'beta' );
-
-		await user.click( screen.getByRole( 'tab', { name: 'Alpha' } ) );
-
-		expect( getSelectedTab() ).toHaveTextContent( 'Alpha' );
-		expect(
-			screen.getByRole( 'tabpanel', { name: 'Alpha' } )
-		).toBeInTheDocument();
-		expect( panelRenderFunction ).toHaveBeenLastCalledWith( TABS[ 0 ] );
-		expect( mockOnSelect ).toHaveBeenLastCalledWith( 'alpha' );
-	} );
+	describe( 'Uncontrolled mode', () => {
```

Comment:
A pattern that we've started to adopt when writing unit tests for components that can be used both in controlled and uncontrolled mode, is to use `describe.each` and run the same suite against the controlled and uncontrolled version.

You can see it in action for the `ComboboxControl` component:

https://github.com/WordPress/gutenberg/blob/56c33cf1f529d286b9c0db49534a2f5617fdcc84/packages/components/src/combobox-control/test/index.js#L80-L83


Of course, if there are certain tests that are specifically aimed at one version of the component, they can be added outside of the `describe.each` call.